### PR TITLE
Parallelize class loading

### PR DESCRIPTION
**Pull request by forestbelton**

### CODE REVIEW

File: framework/src/play/classloading/ApplicationClassloader.java [hunk 1]
```diff
@@ -427,13 +432,35 @@ int computePathHash() {
...
+
+                            return true;
+                        }
+                    });
                 }
 
+                executor.shutdown();
+                try {
+                    while (!executor.awaitTermination(1, TimeUnit.SECONDS)) {
+                        Logger.trace("Waiting for class compiler threads to complete, sleeping 1s...");
```
Comment (jakeyr) [inline framework/src/play/classloading/ApplicationClassloader.java]: Change this logger to say "class loader threads" or "class enhancer threads"

File: framework/src/play/classloading/ApplicationClassloader.java [hunk 2]
```diff
@@ -427,13 +432,35 @@ int computePathHash() {
...
+                    }
+                } catch (Exception e) {}
```
Comment (jakeyr) [inline framework/src/play/classloading/ApplicationClassloader.java]: This seems scary... can we log at least?

Comment (forestbelton) [inline framework/src/play/classloading/ApplicationClassloader.java]: Sure thing, was just copying your pattern from the template loader :wink:

File: framework/src/play/classloading/ApplicationClassloader.java (hunk 1 above)
Comment (forestbelton) [inline framework/src/play/classloading/ApplicationClassloader.java]: Done

**Comment by jakeyr** [comment]
#complete 

:heart:

**Pull request merged**